include/rawpheno.download.form.inc

```diff
@@ -152,7 +167,35 @@ function rawpheno_download($form, &$form_state) {
   );
 
   // Get traits given a location and project.
-  $opt_trait = rawpheno_download_load_traits($location_id, $project_id);
+  if (isset($project_selected) && isset($location)) {
+    $project_id = $project_selected;
+    $location_id = $location;
+  }
+
+  // Extract list of unique traits from rawdata summary materialized view.
+  // Filtered by location and project id number. Default to all location in the default project.
+  $sql = "SELECT ARRAY_TO_STRING(ARRAY_AGG(DISTINCT all_traits), ',') AS all_traits
+          FROM {rawpheno_rawdata_mview}
+          WHERE location IN(:location) AND plant_id IN
+            (SELECT plant_id FROM pheno_plant_project WHERE project_id = :project_id)
+          LIMIT 1";
+
+  $traits = chado_query($sql, array(':location' => $location_id, ':project_id' => $project_id))
+    ->fetchObject();

```

Comment:
You could use `fetchField()` here as well then you wouldn't need to grab them below.